src/settings.ts

```diff
@@ -42,7 +49,11 @@ export async function getSettings() {
   const fileHandle = await open(filePath, 'r')
   try {
     const settings = await fileHandle?.readFile({ encoding: 'utf-8' })
-    return JSON.parse(settings) as AppSettings
+    const currentSettings = JSON.parse(settings) as AppSettings
+    return {
+      ...defaultSettings,
+      ...currentSettings,
+    }
```

Comment:
Ensure new configs that are not saved in the settings file are present in the settings state.